DEVELOPER_GUIDE.md

```diff
@@ -6,9 +6,19 @@ So you want to contribute code to this project? Excellent! We're glad you're her
 - [Submitting Changes](#submitting-changes)
 
 ## Prerequisites
-This project runs as a plugin of [OpenSearch Dashboards](**https://github.com/opensearch-project/OpenSearch-Dashboards) and needs an [OpenSearch server](https://github.com/opensearch-project/OpenSearch) running with the [Security Plugin](https://github.com/opensearch-project/security) installed. At the time of this writing there is a strict version check between these components, so we recommend running all of them from their respective branches with matching versions (this will also ensure they work well together before we cut a new release.)
 
-As a prerequisite, please follow [the developer guide of the Security Plugin](https://github.com/opensearch-project/security/blob/main/DEVELOPER_GUIDE.md). This will leave you with a running OpenSearch server with security enabled. For the sake of this guide, let's assume the latest versions at the time of writing (`1.3.0-SNAPSHOT` for OpenSearch and OpenSearch Dashboards, and `1.3.0.0-SNAPSHOT` for both backend and frontend (this repo) security plugins.) Update the config file (`config/opensearch.yml`) to look like this one:
+This project is a plugin of [OpenSearch-Dashboards](**https://github.com/opensearch-project/OpenSearch-Dashboards). It requires an [OpenSearch](https://github.com/opensearch-project/OpenSearch) server running with the [Security](https://github.com/opensearch-project/security) plugin installed. At the time of this writing there is a strict version check between these components, so we recommend running all of them from their respective branches with matching versions (this will also ensure they work well together before we cut a new release.)
+
+As a prerequisite, please follow [the developer guide of the Security Plugin](https://github.com/opensearch-project/security/blob/main/DEVELOPER_GUIDE.md). This will get a OpenSearch server running with security plugin enabled. 
+
+**IMPORTANT**: Throughout this guide we will be using 1.x branches as our source for all 4 repos (OpenSearch, OpenSearch-Dashboards, security and security-dashboards-plugin).
+
+> NOTE: If you are following this guide by the dot, please make sure that source code that you compile for OpenSearch project using `./gradlew localDistro` is done from [1.x branch](https://github.com/opensearch-project/OpenSearch/tree/1.x)
```

Comment:
Are we updating this statement every time the branch is changed?

Reply:
L14 says this is used just during the guide as just an example. The paragraph above makes reference to the need for matching versions. But yeah, if we move to a major version (2, 3...) it might be a bit confusing so we can update. Perhaps we can omit L14 as L10 covers the general idea and then we move on to use 1.3.0 as a driving example (that part can stay as it is clear it's just for the sake of the example code)

Reply:
I think down the road when there are multiple versions and branches for developers to choose, adding a table to show each version with there corresponding branch will be helpful

Reply:
I have added the relevant branch and version info here... we can add info to this table as we develop in the future